ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -290,17 +314,22 @@ void destroyWindow() { DestroyWindow(enigma::hWnd); }
 void showWindow() { ShowWindow(enigma::hWnd, 1); }
 
 void set_working_directory() {
-    // Set the working_directory
-  WCHAR buffer[MAX_PATH];
-  GetCurrentDirectoryW(MAX_PATH, buffer);
+  // Set the working_directory
+  WCHAR buffer[MAX_PATH + 1];
+  GetCurrentDirectoryW(MAX_PATH + 1, buffer);
   enigma_user::working_directory = shorten(buffer);
 
   // Set the program_directory
-  memset(&buffer[0], 0, MAX_PATH);
-  GetModuleFileNameW(NULL, buffer, MAX_PATH);
+  memset(&buffer[0], 0, MAX_PATH + 1);
```

Comment:
`memset` actually takes the size in bytes, while MAX_PATH gives the number of wchars. This was technically already wrong, but it doesn't matter, because GetModuleFilename should always write the terminating null character. So just setting the first byte to zero should suffice.